### PR TITLE
Add MATICX as collateral in Polygon USDC market

### PR DESCRIPTION
**Pull request by kevincheng96**
MATICX team's proposal to add MATICX as a collateral asset on the Polygon cUSDCv3 market.

### CODE REVIEW

File: scenario/AddMaticxCollateralScenario.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+const MATICX_WHALES = {
+  polygon: ['0x68B9220B8E617b7700aCAE1a5Ff43F3eb29257F3'],
+};
+
+scenario.only(
+  'add new asset maticx',
+  {
+    filter: async (ctx) =>
+      matchesDeployment(ctx, [{ network: 'polygon' }, { network: 'mumbai' }]),
```
Comment (kevincheng96) [inline scenario/AddMaticxCollateralScenario.ts:19]: This won't work for mumbai right? The addresses you used for MATICX and the pricefeed are only for Polygon.

File: scenario/AddMaticxCollateralScenario.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    tokenBalances: {
+      $comet: { $base: '>= 1' },
+    },
+  },
+  async (
+    { comet, configurator, proxyAdmin, actors, bridgeReceiver },
+    context
+  ) => {
+    console.log('DEBUG::0');
```
Comment (kevincheng96) [inline scenario/AddMaticxCollateralScenario.ts:28]: Reminder to remove these logs

**Comment by kevincheng96** [comment]
Cleaned up this PR a bit.

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      (event) => event.event === "ProposalCreated"
+    );
+    const [proposalId] = event.args;
+    trace(`Created proposal ${proposalId}.`);
+  },
+
+  async verify(deploymentManager: DeploymentManager) {
+    const { comet, configurator } = await deploymentManager.getContracts();
+
+    const maticxAssetIndex = 3; // TODO
```
Comment (kevincheng96) [inline deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:104]: The index should just be `comet.numAssets() - 1`.

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    const mainnetActions = [
+      // 1. Set Comet configuration and deployAndUpgradeTo new Comet on Polygon.
+      {
+        contract: fxRoot,
+        signature: "sendMessageToChild(address,bytes)",
+        args: [bridgeReceiver.address, l2ProposalData],
+      },
+    ];
+
+    const description = "TODO";
```
Comment (kevincheng96) [inline deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:87]: TODO

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts (hunk 3 above)
Comment (manoj9april) [reply deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:104]: This is NaN.

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
...n Stader on Polygon protocol contracts are dual audited:\n\n [Here](https://staderlabs-docs.s3.amazonaws.com/audits/polygon/StaderLabs_MaticX_Smart_Contract_Security_Audit_Report_Halborn_Final.pdf) is the link to the Audit completed by Halborn\n [Here](https://staderlabs-docs.s3.amazonaws.com/audits/polygon/StaderLabs_maticX_Audit_Report_Immunebytes.pdf) is the link to the Audit completed by Immunebytes\n Stader's contracts for MaticX are controlled by a multi-sig account (0x91B4139A2FAeaCD4CdbFc3F7B1663F91a54be237) managed by the internal as well as external parties. The confirmation count is 3 out of 5 signatures required $1Mn Bug Bounty on [Immunefi](https://immunefi.com/bounty/StaderforPolygon/)\n Ongoing monitoring and on-chain security tracking by [Forta](https://app.forta.network/agents/stader-labs) External multi-sig and time-lock drive the staking contract\n\n ## References\n\n [MaticX Chainlink Oracle Price Feed](https://polygonscan.com/address/0x5d37E4b374E6907de8Fc7fb33EE3b0af403C7403)\n [Polygonscan - MaticX address](https://polygonscan.com/token/0xfa68fb4628dff1028cfec22b4162fccd0d45efb6)\n [Forum Discussion](https://www.comp.xyz/t/listing-maticx-on-compound/4306)\n";
```
Comment (manoj9april) [inline deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:87]: can we visualize it somehow?

Comment (kevincheng96) [reply deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:87]: The only way is to make this proposal on testnet and view it on https://compound.finance/governance?target_network=goerli.

Comment (kevincheng96) [reply deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:87]: Yeah, we can help create one on testnet to sanity check the description when this proposal is ready.

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts (hunk 3 above)
Comment (kevincheng96) [reply deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts]: Updated

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts (hunk 5 above)
Comment (manoj9april) [reply deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:87]: cool, thanks!

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    return {};
+  },
+
+  enact: async (
+    deploymentManager: DeploymentManager,
+    govDeploymentManager: DeploymentManager,
+    vars: Vars
+  ) => {
+    const trace = deploymentManager.tracer();
+    // const ethers = deploymentManager.hre.ethers; // ethers is not available
```
Comment (scott-silver) [inline deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:23]: nit: delete commented-out lines

File: scenario/AddMaticxCollateralScenario.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  },
+  async (
+    { comet, configurator, proxyAdmin, actors, bridgeReceiver },
+    context
+  ) => {
+    const { albert } = actors;
+    const dm = context.world.deploymentManager;
+    const maticx = await dm.existing(
+      'MATICX',
+      '0xfa68FB4628DFF1028CFEc22b4162FCcd0d45efb6',
```
Comment (scott-silver) [inline scenario/AddMaticxCollateralScenario.ts:32]: nit: maybe move this address into a constant

File: scenario/AddMaticxCollateralScenario.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+      context.world.base.network,
+      'contracts/ERC20.sol:ERC20'
+    );
+    const maticxPricefeed = await dm.existing(
+      'MATICX:priceFeed',
+      '0x5d37E4b374E6907de8Fc7fb33EE3b0af403C7403',
```
Comment (scott-silver) [inline scenario/AddMaticxCollateralScenario.ts:38]: same with this address

File: .github/workflows/run-scenarios.yaml [hunk 9]
```diff
@@ -2,6 +2,12 @@ name: Run Scenarios
 on:
   workflow_dispatch:
   pull_request:
+permissions:
```
Comment (scott-silver) [inline .github/workflows/run-scenarios.yaml:5]: why do we need added permissions for this action?

Comment (manoj9april) [reply .github/workflows/run-scenarios.yaml:5]: @kevincheng96

File: scenario/AddMaticxCollateralScenario.ts (hunk 8 above)
Comment (manoj9april) [reply scenario/AddMaticxCollateralScenario.ts:38]: sure

File: scenario/AddMaticxCollateralScenario.ts (hunk 7 above)
Comment (manoj9april) [reply scenario/AddMaticxCollateralScenario.ts:32]: sure

File: deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts (hunk 6 above)
Comment (manoj9april) [reply deployments/polygon/usdc/migrations/1689168483_add_maticx_collateral.ts:23]: sure

File: .github/workflows/run-scenarios.yaml (hunk 9 above)
Comment (kevincheng96) [reply .github/workflows/run-scenarios.yaml:5]: @manoj9april was getting [this error](https://github.com/dorny/test-reporter/issues/168) when running scenarios in his own forked repo, so this was the way to address that. We don't need the explicit permissions in the `comet` repo, but other teams that are developing it in their own repos might.

Comment (manoj9april) [reply .github/workflows/run-scenarios.yaml:5]: cool will remove from this PR then

**Comment by manoj9april** [comment]
@kevincheng96 please review this https://github.com/compound-finance/comet/pull/796

**Comment by manoj9april** [comment]
@kevincheng96 Please review this as well https://github.com/compound-finance/comet/pull/798
contains only description changes

**Comment by kevincheng96** [comment]
This proposal was executed: https://compound.finance/governance/proposals/172

**Review by scott-silver — APPROVED** [review]

**Review by selarun13 — COMMENTED** [review]
0x6F99E90Dd385c23826b416528e325214671FeBf6

**Review by selarun13 — COMMENTED** [review]
0x6F99E90Dd385c23826b416528e325214671FeBf6

File: scenario/LiquidationBotScenario.ts [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import { scenario } from './context/CometContext';
```
Comment (selarun13) [inline scenario/LiquidationBotScenario.ts:1]: 0x6F99E90Dd385c23826b416528e325214671FeBf6

**Review by selarun13 — COMMENTED** [review]
0x6F99E90Dd385c23826b416528e325214671FeBf6

File: scenario/LiquidationBotScenario.ts (hunk 10 above)
Comment (selarun13) [inline scenario/LiquidationBotScenario.ts:1]: 0x6F99E90Dd385c23826b416528e325214671FeBf6

**Review by selarun13 — COMMENTED** [review]
0x6F99E90Dd385c23826b416528e325214671FeBf6

**Review by selarun13 — COMMENTED** [review]
0x6F99E90Dd385c23826b416528e325214671FeBf6

**Pull request merged**